### PR TITLE
11044 edit facet bug

### LINKED ISSUE #11044: Multiple listing for facets after editing the list
**What steps does it take to reproduce the issue?**
I was testing other things when I went to a collection's Edit/General Info form and added an additional facet, and then added another one. Looking at the list of existing facets in the form, I began to see doubles (and at QDR, even quadruples). 

image

This appears to happen on beta.dataverse.org (JSF UI) but not demo.dataverse.org, so I'm guessing it is a problem introduced post v6.4. (My QDR test machine has code merged from develop just a couple days ago.

At QDR, I verified that duplicates are in the database, e.g. multiple entries in dataversefacet for the same datasetfieldtype_id and dataverse_id: 
image

I'm not sure what is happening with the display order.

* When does this issue occur?

* Which page(s) does it occurs on?

* What happens?

* To whom does it occur (all users, curators, superusers)?

* What did you expect to happen?

**Which version of Dataverse are you using?**
develop branch post 6.4

**Any related open or closed issues to this bug report?**

**Screenshots:**

No matter the issue, screenshots are always welcome.

To add a screenshot, please use one of the following formats and/or methods described here:

* https://help.github.com/en/articles/file-attachments-on-issues-and-pull-requests
*

**Are you thinking about creating a pull request for this issue?**
Help is always welcome, is this bug something you or your organization plan to fix?

### PR DESCRIPTION
**Pull request by sekmiller**
**What this PR does / why we need it**:
Fixes an issue where the facets would be added multiple times on Edit Dataverse

**Which issue(s) this PR closes**:

- Closes #11044 Multiple listing of Facets after Dataverse Edit

**Special notes for your reviewer**:
The issue was that while the existing facets were deleted in the DB, they were left as part of the Dataverse being edited so when the dataverse is saved the facets were re-created. Similar situation for field input levels.

**Suggestions on how to test this**: Edit Dataverse via api and UI and verify that the facets and dataset field input levels are being saved correctly with no doubling of entries. (see original ticket for example of failure state.)

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**: No

**Additional documentation**: None - fixing documented behavior

### CODE REVIEW

**Comment by qqmyers** [comment]
Haven't tried to review - would it possible to also add a db constraint to avoid more than one entry per datasetfieldtype_id + dataverse_id combo?

**Comment by sekmiller** [comment]
@qqmyers I thought about adding a DB constraint, but I was worried that if it was violated it would fail badly - or at least in a way that would be difficult for the end user to understand/fix.

File: src/main/java/edu/harvard/iq/dataverse/DataverseFacetServiceBean.java [hunk 1]
```diff
@@ -42,23 +42,24 @@ public void delete(DataverseFacet dataverseFacet) {
...
-        
+
         dataverseFacet.setDisplayOrder(displayOrder);
         dataverseFacet.setDatasetFieldType(fieldType);
         dataverseFacet.setDataverse(ownerDv);
-        
-        ownerDv.getDataverseFacets().add(dataverseFacet);
+
         em.persist(dataverseFacet);
+        ownerDv.getDataverseFacets().add(dataverseFacet);
```
Comment (qqmyers) [inline src/main/java/edu/harvard/iq/dataverse/DataverseFacetServiceBean.java:61]: Just curious - did this have to move after the persist?

**Review by qqmyers — APPROVED** [review]
Looks good - I haven't tested but the changes seem reasonable given the issue. @sekmiller - there are two failing tests though. Perhaps the deleteFile one is just timing but the other looks like it's facet-related. I'll approve, conditional on it being able to pass the tests before QA completes.

File: src/main/java/edu/harvard/iq/dataverse/DataverseFacetServiceBean.java (hunk 1 above)
Comment (sekmiller) [reply src/main/java/edu/harvard/iq/dataverse/DataverseFacetServiceBean.java:59]: maybe not. I put it back and it still seems to work

**Comment by ofahimIQSS** [comment]
Passed testing - Merging PR 
[Testing of 11052.docx](https://github.com/user-attachments/files/18010231/Testing.of.11052.docx)

**Pull request merged**